scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -128,12 +140,76 @@ public SchedulingService(SchedulingInfrastructure infrastructure, SchedulerState
 
         pinger = new NodePingThread(this);
         pinger.start();
+
+        if (PASchedulerProperties.SCHEDULER_AUTOMATIC_REMOVED_JOB_DELAY.getValueAsInt() > 0) {
+            startHouseKeeping(this.jobsToDeleteFromDB, this.infrastructure);
+        }
+    }
+
+    public void startHouseKeeping(final ConcurrentLinkedQueue<JobId> jobsQueue,
```

Comment:
seems that jobsQueue not used at all, so that jobsToDeleteFromDB field seems useless !